src/GameData.ts

```diff
@@ -0,0 +1,262 @@
+
+    public async loadItemData(): Promise<ItemData[]> {
+        const items = await fetch(this.sharedSettings.lookup.itemUrl).then(x => x.json()) as ItemData[];
+        for (const item of items) {
+            const other = items.find(x => x.id === item.id)!;
+
+            other.from = item.from.map(x => items.find(y => y.id === +x)!).filter(x => x).map(x => x.name);
+            other.to = item.to.map(x => items.find(y => y.id === +x)!).filter(x => x).map(x => x.name);
+        }
+
+        return items;
+    }
+
+    public onLookup(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        const supportedTypes = ["item", "perk", "rune", "champion"];
+
+        if (args.length === 0) {
+            const response = `Usage: !${command} [type] [term]. Supported types are ` + supportedTypes.map(x => "`" + x + "`").join(", ");
+            message.channel.send(response);
+            return;
+        }
+
+        if (!supportedTypes.includes(args[0])) {
+            message.channel.send(`I'm sorry. I'm unable to parse the category \`${args[0]}\` at this moment. If you think it should be added, please contact a guru.`);
+            return;
+        }
+
+        let result: string | any[] = [];
+        switch (args[0]) {
+            case "item": {
+                result = this.findItem(args.slice(1).join(" "));
+                break;
+            }
+            case "perk": // fall through
+            case "rune": {
+                result = this.findPerk(args.slice(1).join(" "));
+                break;
+            }
+            case "champion": {
+                result = this.findChampion(args.slice(1).join(" "));
+                break;
+            }
+        }
+
+        if (typeof result === "string") {
+            message.channel.send(result);
+            return;
+        }
+
+        result.forEach(x => message.channel.send("```" + JSON.stringify(x, null, 4) + "```"));
+    }
+
+    public findItem(search: string): string | any[] {
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        // match name loosely, but ids strictly
+        const result = this.itemData.map(c => ({
+            value: c,
+            textscore: Math.min(
+                levenshteinDistance(search, c.name),
+                levenshteinDistanceArray(search, c.categories),
+            ),
+            idscore: Math.min(
+                levenshteinDistance(search, c.id.toString()),
+            ),
+        })).filter(x => x.textscore < this.sharedSettings.lookup.textConfidence || x.idscore < this.sharedSettings.lookup.numberConfidence);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        result.sort((a, b) => Math.min(a.textscore, a.idscore) > Math.min(b.textscore, b.idscore) ? 1 : -1);
+
+        if (result.length > this.sharedSettings.lookup.returnedEntryCount) {
```

Comment:
Don't you agree Annie would be on top of this list?
![image](https://user-images.githubusercontent.com/3397066/55379803-73b2fa80-551e-11e9-82be-594afc8017f4.png)

Reply:
There's a bug with this function: 
![image](https://user-images.githubusercontent.com/3397066/55379881-b674d280-551e-11e9-9d03-73a102ec28af.png)

I think any ID returns Vi as well.

Reply:
Thats a result of the textsearch having a value of 2, since vi only has 2 chars, she always matches.. 
I'll make some changes to it when i come home to fix it.